From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Mon, 26 Mar 2018 18:30:53 +0300
Subject: [PATCH] Make player data saving configurable

Upstream has added a patch which negates the need for this patch,
however, we should still migrate our configuration back upstream,
to prevent unexpected situations

diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index b020a5d69c2d6a55577c0fdcfc2a3ab1e35bcf36..2e90709bdd42b4dd5394c977b45b427047713ba5 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -280,4 +280,13 @@ public class PaperConfig {
     private static void authenticationServersDownKickMessage() {
         authenticationServersDownKickMessage = Strings.emptyToNull(ChatColor.translateAlternateColorCodes('&', getString("messages.kick.authentication-servers-down", authenticationServersDownKickMessage)));
     }
+
+    private static void savePlayerData() {
+        Object val = config.get("settings.save-player-data");
+        if (val instanceof Boolean) {
+            SpigotConfig.disablePlayerDataSaving = !(Boolean) val;
+            SpigotConfig.config.set("players.disable-saving", SpigotConfig.disableAdvancementSaving);
+            SpigotConfig.save();
+        }
+    }
 }
